docs/content/guides/dagster-pipes/integrating-databricks-with-dagster-pipes.mdx

```diff
@@ -1,6 +1,305 @@
+
+   ```python
+   # dagster_databricks_pipes.py
+
+   from dagster import AssetExecutionContext, Definitions, EnvVar, asset
+   from dagster_databricks import PipesDatabricksClient
+   from databricks.sdk.service import jobs
+   ```
+
+3. Next, you’ll define the asset. Copy and paste the following below the imports in `dagster_databricks_pipes.py`:
+
+   ```python
+   # dagster_databricks_pipes.py
+
+   @asset
+   def databricks_asset(context: AssetExecutionContext, pipes_databricks: PipesDatabricksClient):
+       task = jobs.SubmitTask.from_dict({
+           "new_cluster": { ... },
+           "libraries": [
+               # must include dagster-pipes
+               {"pypi": {"package": "dagster-pipes"}},
+           ],
+           "task_key": "some-key",
+           "spark_python_task": {
+               "python_file": "dbfs:/existing_databricks_code.py",   # location of target code file
+               "source": jobs.Source.WORKSPACE,
+           }
+       })
+   ```
+
+   Here’s what we did in this example:
+
+   - Created an asset named `databricks_asset`
+   - Provided <PyObject object="AssetExecutionContext" /> as the `context` argument to the asset. This object provides system information such as resources, config, and logging. We’ll come back to this a bit later in this section.
+   - Specified a resource for the asset to use, <PyObject module="dagster_databricks" object="PipesDatabricksClient" />. We’ll also come back to this in a little bit.
+   - Defined a Databricks `SubmitTask` specification in the asset body. When the asset is materialized, this specification will be passed to the Databricks resource (<PyObject module="dagster_databricks" object="PipesDatabricksClient" />), which will use it to launch a Databricks job.
+
+     - `new_cluster` - TODO
+
+       - `spark_env_vars` - **Optional.** An object of key-value pairs specifying environment variables to inject into the launched Databricks job.
+
+     - `libraries` - An object containing the libraries required to run the specified Python file (`spark_python_task.python_file`). This object must include `dagster-pipes`; for example:
+
+       ```python
+       "libraries": [
+          {
+          "pypi": {
+           "package": "dagster-pipes"
+          }
+         },
+       ],
+       ...
+       ```
+
+     - `spark_python_task` - An object specifying the Python file the job should run, which contains the following properties:
+       - `python_file` - The URI of the Python file to run, located in DBFS.
+       - `source` - TODO - is this required?
+
+### Step 1.2: Create the Databricks job
+
+In this step, we’ll create the Databricks job and pass in the task specification, as well as our `context` data. The Databricks resource used by the asset contains a `run` method. This method synchronously executes the Databricks job using the task specification you created in [the previous step](#step-11-define-a-dagster-asset).
```

Comment:
- "contains a `run` method" -> "exposes a `run` method"
- "using the task specification" -> "defined by the task specification"